File: flipped_material/python/code/guessing_game/guessing_game_4.py
from random import randint
min_value = 1
max_value = 100
answer = randint(min_value, max_value)
guess = -1
moves = 0
while answer != guess:
    guess = int(input(f"Guess a number between {min_value} and {max_value} "))
    
    if guess < min_value or guess > max_value:
        continue

    moves = moves + 1
    if guess < answer:
        print("your guess was too low!")
    elif guess > answer:
        print("your guess was too high!")
print(f"{guess} was the right answer. You won in {moves} moves.")

File: .old/2022/flipped_material/cpp/code/guessing_game/guessing_game4.c
#include <stdio.h>

int main(void) {
  int answer = 5;
  int guess = -1;
  int min_value = 1;
  int max_value = 10;
  
  while (answer != guess){
    printf("pick a number between %d and %d\n", min_value, max_value);
    scanf("%d", &guess);
    if (guess < min_value || guess > max_value) {
      continue;
    }
    if (guess == answer){
      printf("You win!");
    } else {
      if (guess < answer){
        printf("Your guess was too low!\n");
      } else {
        printf("Your guess was too high!\n");
      }
    }
  }
  return 0;
}